libyul/AST.h

```diff
@@ -69,14 +73,20 @@ class LiteralValue {
 struct Literal { langutil::DebugData::ConstPtr debugData; LiteralKind kind; LiteralValue value; };
 /// External / internal identifier or label reference
 struct Identifier { langutil::DebugData::ConstPtr debugData; YulName name; };
+/// Builtin function
+struct Builtin { langutil::DebugData::ConstPtr debugData; BuiltinHandle handle; };
+/// Verbatim function
+struct Verbatim { langutil::DebugData::ConstPtr debugData; VerbatimHandle handle; };
```

Comment:
When I saw this I immediately started wondering why verbatim is so special that we need to be aware of it at the AST level. It does not seem like it belongs here.

Apparently the answer is that the verbatim functions don't exist by default and are created by the dialect on demand. But I think it could be solved differently. There are at most `99 * 99` possible functions and the IDs don't have to be contiguous. We could have `EVMDialect` reserve a specific range for their IDs and handle them uniformly with other builtins. This way we wouldn't even have to expose then in `Dialect`, it would stay an implementation detail of this specific dialect.

The IDs are also not exposed to the user in the AST so if this turns out to be too restrictive, we can still change it. But for now this seems to me like a better approach.

Reply:
I wasn't aware it could "only" be 10000 of them. In that case, sure, we can handle it uniformly. Doesn't make the dialect implementation any nicer, though.

Reply:
It is! Removed lots of duplicate code branches, it's a definite win. In the end it _is_ a builtin function, so we should treat it as such. I just wasn't aware that there was a hard limit (beyond size_t) on it, in which case offsetting the ids is difficult :)